server/src/test/java/io/crate/expression/scalar/HasTablePrivilegeFunctionTest.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Licensed to Crate.io GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.expression.scalar;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.List;
+import java.util.Set;
+
+import org.junit.Test;
+
+import io.crate.metadata.RelationName;
+import io.crate.metadata.Schemas;
+import io.crate.metadata.pgcatalog.OidHash;
+import io.crate.role.Permission;
+import io.crate.role.Policy;
+import io.crate.role.Privilege;
+import io.crate.role.Role;
+import io.crate.role.Securable;
+import io.crate.role.metadata.RolesHelper;
+import io.crate.testing.SqlExpressions;
+
+public class HasTablePrivilegeFunctionTest extends ScalarTestCase {
+
+    private static final Role TEST_USER_WITH_SYS_SUMMITS_TABLE_DQL =
+        RolesHelper.userOf("testUserWithSysSummitsDQL", Set.of(
+                new Privilege(Policy.GRANT, Permission.DQL, Securable.TABLE, "sys.summits", Role.CRATE_USER.name())),
+            null);
+
+    private static final Role TEST_USER_WITH_SYS_HEALTH_TABLE_DQL =
+        RolesHelper.userOf("testUserWithSysHealthDQL", Set.of(
+                new Privilege(Policy.GRANT, Permission.DQL, Securable.TABLE, "sys.cluster", Role.CRATE_USER.name())),
+            null);
+
+    private static final Role TEST_USER_WITH_DOC_USERS_TABLE_DQL =
+        RolesHelper.userOf("testUserWithUsersTableDQL", Set.of(
+                new Privilege(Policy.GRANT, Permission.DQL, Securable.TABLE, "doc.users", Role.CRATE_USER.name())),
```

Comment:
Minor, but maybe add a DML here to test also a `true` return for another permission like ``update`` or so.